packages/api/amplify_api/lib/src/api_plugin_impl.dart

```diff
@@ -0,0 +1,397 @@
+        OidcFunctionAuthProvider(authProvider),
+      );
+    }
+  }
+
+  @override
+  Future<void> addPlugin({
+    required AmplifyAuthProviderRepository authProviderRepo,
+  }) async {
+    await super.addPlugin(authProviderRepo: authProviderRepo);
+
+    if (zIsWeb || !(Platform.isAndroid || Platform.isIOS)) {
+      return;
+    }
+
+    // Configure this plugin to act as a native iOS/Android plugin.
+    final nativePlugin = _NativeAmplifyApi(_authProviders);
+    NativeApiPlugin.setup(nativePlugin);
+
+    final nativeBridge = NativeApiBridge();
+    try {
+      final authProvidersList =
+          _authProviders.keys.map((key) => key.rawValue).toList();
+      await nativeBridge.addPlugin(authProvidersList);
+    } on PlatformException catch (e) {
+      if (e.code == 'AmplifyAlreadyConfiguredException' ||
+          e.code == 'AlreadyConfiguredException') {
+        throw const AmplifyAlreadyConfiguredException(
+          AmplifyExceptionMessages.alreadyConfiguredDefaultMessage,
+          recoverySuggestion:
+              AmplifyExceptionMessages.alreadyConfiguredDefaultSuggestion,
+        );
+      }
+      throw AmplifyException.fromMap((e.details as Map).cast());
+    }
+  }
+
+  /// Returns the HTTP client to be used for REST/GraphQL operations.
+  ///
+  /// Use [apiName] if there are multiple endpoints of the same type.
+  @visibleForTesting
+  AWSHttpClient getHttpClient(
+    EndpointType type, {
+    String? apiName,
+    APIAuthorizationType? authorizationMode,
+  }) {
+    final endpoint = _apiConfig.getEndpoint(
+      type: type,
+      apiName: apiName,
+    );
+    final authModeForClientKey =
+        authorizationMode ?? endpoint.config.authorizationType;
+    final clientPoolKey = '${endpoint.name}.${authModeForClientKey.name}';
+    return _clientPool[clientPoolKey] ??= AmplifyHttpClient(
+      baseClient: AmplifyAuthorizationRestClient(
+        endpointConfig: endpoint.config,
+        baseClient: _baseHttpClient,
+        authorizationMode: authorizationMode,
+        authProviderRepo: _authProviderRepo,
+      ),
+    )..supportedProtocols = SupportedProtocols.http1;
+  }
+
+  /// Returns the websocket connection to use for a given endpoint.
+  ///
+  /// Use [apiName] if there are multiple endpoints.
+  @visibleForTesting
+  WebSocketConnection getWebSocketConnection({String? apiName}) {
+    final endpoint = _apiConfig.getEndpoint(
+      type: EndpointType.graphQL,
+      apiName: apiName,
+    );
+    return _webSocketConnectionPool[endpoint.name] ??= WebSocketConnection(
+      endpoint.config,
+      _authProviderRepo,
+      subscriptionOptions: subscriptionOptions,
+      logger: _logger.createChild(
+        'webSocketConnection${endpoint.name}',
+      ),
+    );
+  }
+
+  Uri _getGraphQLUri(String? apiName) {
+    final endpoint = _apiConfig.getEndpoint(
+      type: EndpointType.graphQL,
+      apiName: apiName,
+    );
+    return endpoint.getUri(path: null, queryParameters: null);
```

Comment:
Don't think this adds much 


```suggestion
    return endpoint.getUri();
```